yt-playlists-delete-enhancer.js

```diff
@@ -159,10 +159,19 @@ class GMScript {
     async handleRemoveVideosClickedEvent(watchTimeValue) {
         this.disableRemoveButton();
         let idsToDelete = this.getVideosIdsToDelete(watchTimeValue, this.playlistVideos);
-        const respjson = await this.removeVideosFromPlaylist(this.playlistName, idsToDelete);
-        if (respjson.status === "STATUS_SUCCEEDED") {
-            // TODO propagate the change directly to YT UI instead of reloading the all page
-            location.reload();
+      
+        const videoIdMap = this.playlistVideos.reduce((acc, curr) => {
```

Comment:
You want to create this array only if there are ids to delete and if the remove request succeeded. Since there is no use for it if either of those conditions fail.

You could move this line inside the block l.171.